tests/contest/contest/src/tests/cgroups/devices.rs

```diff
@@ -0,0 +1,84 @@
+use std::path::Path;
+
+use anyhow::Context;
+use oci_spec::runtime::{
+    LinuxBuilder, LinuxDeviceCgroup, LinuxDeviceCgroupBuilder, LinuxDeviceType,
+    LinuxResourcesBuilder, Spec, SpecBuilder,
+};
+use test_framework::{test_result, ConditionalTest, TestGroup, TestResult};
+
+use crate::utils::test_outside_container;
+use crate::utils::test_utils::check_container_created;
+
+fn can_run() -> bool {
+    Path::new("/sys/fs/cgroup/devices").exists()
+}
+
+fn linux_device_build(
+    allow: bool,
+    dev_type: LinuxDeviceType,
+    major: i64,
+    minor: i64,
+    access: String,
+) -> LinuxDeviceCgroup {
+    LinuxDeviceCgroupBuilder::default()
+        .access(allow.to_string())
+        .typ(dev_type)
+        .major(major)
+        .minor(minor)
+        .access(access)
+        .build()
+        .unwrap()
+}
+
+fn create_spec(cgroup_name: &str, devices: Vec<LinuxDeviceCgroup>) -> anyhow::Result<Spec> {
+    let spec = SpecBuilder::default()
+        .linux(
+            LinuxBuilder::default()
+                .cgroups_path(Path::new("/runtime-test").join(cgroup_name))
```

Comment:
I think this is incorrect. For absolute cgroups path, it should be single level at root like `/cgrouptest` . Joining another path here would make it relative.